update.sh

```diff
@@ -2,14 +2,19 @@
 
 set -e
 
-if [[ ! $# -eq 2 ]]; then
-	echo "usage: $0 [ruby|mruby|jruby|rubinius|truffleruby|truffleruby-graalvm] [VERSION]"
+if [[ $# -lt 2 ]]; then
+	echo "usage: $0 [ruby|mruby|jruby|rubinius|truffleruby|truffleruby-graalvm] [VERSION] [RELEASE_DIRECTORY]"
 	exit 1
 fi
 
 ruby="$1"
 version="$2"
-dest="pkg"
+dest="$3"
+if [ -z "$dest" ]; then
+	dest="pkg"
+fi
+
+root=$(pwd)
```

Comment:
A safer method would be `${0%/*}/..` which would ensure everything is relative to the location of `update.sh`.

Reply:
I tried it but that doesn't seem to work, it returns `./..` which is not an absolute path but that's needed to stay correct after `pushd` below.

Reply:
`root="$(dirname "$(realpath -s "$0")")"` works, should I switch to that? (from https://stackoverflow.com/a/11114547/388803)
I'm not sure `realpath` is available by default on macOS.

Reply:
[Greg's Bash Wiki recommends](https://mywiki.wooledge.org/BashFAQ/028) using `${BASH_SOURCE%/*}` combined with `cd` or `$PWD`. Doing a `cd ${BASH_SOURCE%/*}` should work, which would allow you to run `../update.sh` from within a sub-directory, and the script would ensure all file access is relative to the root directory of the repo wrt the current working directory. So even if `root` ends up being `.` or `..` it should evaluate correctly given the current working directory.

Reply:
In the current script structure absolute paths are needed. I changed the script structure, i.e., I used the pushd/popd just around the part needing it and then we can use relative paths.
TBH I think there is no point to run this script from a PWD != than ruby-versions repo, it did not support it before either.